### PR TITLE
test: replace chai-spies with sinon

### LINKED ISSUE #373: chai-spies fails after upgrading to chai@4
See https://circleci.com/gh/plotly/falcon-sql-client/3785

### PR DESCRIPTION
**Pull request by n-riesco**
* Fixed failures in CircleCI caused by `chai-spies`.

* I tried to use sinon-chai but it throws the same error as chai-spies.

Fixes #373 

---

@tarzzz would you review this PR?

### CODE REVIEW

**Comment by n-riesco** [comment]
@tarzzz I've added a fix for the test failures in the specs for elasticsearch.

**Comment by tarzzz** [comment]
👍  I will give it a look tomorrow.. !!

File: test/backend/QueryScheduler.spec.js [hunk 1]
```diff
@@ -86,28 +85,28 @@ describe('QueryScheduler', function() {
         queryScheduler.job = spy1;
         queryScheduler.scheduleQuery(query);
 
-        expect(spy1).to.not.have.been.called();
+        assert(spy1.notCalled, 'job1 should not have been called yet');
 
         return wait(3.25 * refreshInterval * 1000)
             .then(() => {
-                expect(spy1).to.have.been.called.exactly(3);
+                assert(spy1.calledThrice, 'job1 should have been called yet three times');
```
Comment (tarzzz) [inline test/backend/QueryScheduler.spec.js]: job1 should have been called ~~yet~~~ three times

File: test/backend/QueryScheduler.spec.js [hunk 2]
```diff
@@ -86,28 +85,28 @@ describe('QueryScheduler', function() {
...
 
                 queryScheduler.job = spy2;
                 queryScheduler.scheduleQuery(merge(query, {query: 'query-2'}));
             })
             .then(() => wait(3.25 * refreshInterval * 1000))
             .then(() => {
-                expect(spy1).to.have.been.called.exactly(3);
-                expect(spy1).to.have.been.called.always.with.exactly(
+                assert(spy1.calledThrice, 'job1 should have been called yet three times');
```
Comment (tarzzz) [inline test/backend/QueryScheduler.spec.js]: job1 should have been called ~~~yet~~~ three times

File: test/backend/QueryScheduler.spec.js [hunk 3]
```diff
@@ -86,28 +85,28 @@ describe('QueryScheduler', function() {
...
+                assert(spy1.alwaysCalledWith(
                     query.fid, query.uids, query.query,
                     query.connectionId, query.requestor
-                );
+                ), `job1 was called with unexpected args: ${spy1.args}`);
 
-                expect(spy2).to.have.been.called.exactly(3);
-                expect(spy2).to.have.been.called.always.with.exactly(
+                assert(spy2.calledThrice, 'job2 should have been called yet three times');
```
Comment (tarzzz) [inline test/backend/QueryScheduler.spec.js]: job2 should have been called ~~~yet~~~ three times

**Pull request merged**